WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Service/BlogDashboardPostsParser.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+import CoreData
+
+class BlogDashboardPostsParser {
+    private let managedObjectContext: NSManagedObjectContext
+
+    init(managedObjectContext: NSManagedObjectContext) {
+        self.managedObjectContext = managedObjectContext
+    }
+
+    /// Parse the posts data that comes from the API
+    /// We might run into cases where the API returns that there are no
+    /// drafts and/or scheduled posts, however the user might have
+    /// they locally.
+    /// This parser basically looks in the local content and fixes the data
+    /// if needed.
+    func parse(_ postsDictionary: NSDictionary, for blog: Blog) -> NSDictionary {
+        guard let posts = postsDictionary.mutableCopy() as? NSMutableDictionary else {
+            return postsDictionary
+        }
+
+        if !posts.hasDrafts {
```

Comment:
My understanding here that we're mssing the case where the API returns draft posts but the user has extra draft posts locally. Or am I missing something?

Reply:
Aaah, okay got it 👍